src/hiscores.ts

```diff
@@ -284,91 +285,81 @@ export async function getStats(
   ];
   const shouldGetFormattedRsn = options?.shouldGetFormattedRsn ?? true;
 
-  const mainRes = await httpGet<HiscoresResponse>(
-    getStatsURL('main', rsn, true),
-    options?.axiosConfigs?.main
-  );
-  if (mainRes.status === 200) {
-    const emptyResponse: AxiosResponse<HiscoresResponse> = {
-      status: 404,
-      data: { skills: [], activities: [] },
-      statusText: '',
-      headers: {},
-      config: {} as InternalAxiosRequestConfig
-    };
-    const getModeStats = async (
-      mode: Extract<Gamemode, 'ironman' | 'hardcore' | 'ultimate'>
-    ): Promise<AxiosResponse<HiscoresResponse>> =>
-      otherGamemodes.includes(mode)
-        ? httpGet<HiscoresResponse>(
-            getStatsURL(mode, rsn, true),
-            options?.axiosConfigs?.[mode]
-          ).catch((err) => err)
-        : emptyResponse;
-    const formattedName = shouldGetFormattedRsn
-      ? await getRSNFormat(rsn, options?.axiosConfigs?.rsn).catch(
-          () => undefined
-        )
-      : undefined;
-
-    const player: Player = {
-      name: formattedName ?? rsn,
-      mode: 'main',
-      dead: false,
-      deulted: false,
-      deironed: false
-    };
-    player.main = parseJsonStats(mainRes.data);
-
-    const ironRes = await getModeStats('ironman');
-    if (ironRes.status === 200) {
-      player.ironman = parseJsonStats(ironRes.data);
-      const hcRes = await getModeStats('hardcore');
-      const ultRes = await getModeStats('ultimate');
-      if (hcRes.status === 200) {
-        player.mode = 'hardcore';
-        player.hardcore = parseJsonStats(hcRes.data);
-        if (
-          player.ironman.skills.overall.xp !== player.hardcore.skills.overall.xp
-        ) {
-          player.dead = true;
-          player.mode = 'ironman';
-        }
-        if (
-          player.main.skills.overall.xp !== player.ironman.skills.overall.xp
-        ) {
-          player.deironed = true;
-          player.mode = 'main';
-        }
-      } else if (ultRes.status === 200) {
-        player.mode = 'ultimate';
-        player.ultimate = parseJsonStats(ultRes.data);
-        if (
-          player.ironman.skills.overall.xp !== player.ultimate.skills.overall.xp
-        ) {
-          player.deulted = true;
-          player.mode = 'ironman';
-        }
-        if (
-          player.main.skills.overall.xp !== player.ironman.skills.overall.xp
-        ) {
-          player.deironed = true;
-          player.mode = 'main';
-        }
-      } else {
+  const main = await getOfficialStats(rsn, 'main', options?.axiosConfigs?.main);
+
+  const emptyResponse: HiscoresResponse = {
+    skills: [],
+    activities: []
+  };
+  const getModeStats = async (
+    mode: Extract<Gamemode, 'ironman' | 'hardcore' | 'ultimate'>
+  ): Promise<HiscoresResponse | undefined> =>
+    otherGamemodes.includes(mode)
+      ? getOfficialStats(rsn, mode, options?.axiosConfigs?.[mode])
+      .catch(err => err)
```

Comment:
Similarly to the comment for the `emptyResponse`, we need to return `undefined` in the case of a caught error. Otherwise the conditions of the if statements below will be true when the error object is present as the response.

We should probably log the error in some way, or maybe even consider returning the errors in some format. But on the other hand I'm also hesitant to modify the structure of the `Player` object. 🤔 Not entirely sure what to do here, but since we're updating errors in the PR, it seems like a good opportunity to improve error handling in general.

Reply:
Good point. I wonder what the expected behavior would be for the case where the initial stat lookup succeeds, but there is a `500` status code during the subsequent lookups. We could add an optional `strict` parameter that throws if any of the lookups error.

I will check what the previous behavior would have been prior to the JSON endpoint update. Realistically, the only error that would occur is `HiScoresError`, unless the account in question name changes or gets banned in-between the first lookup and one of the gamemode lookups.

Reply:
I like the idea of a `strict` mode! I think that if `getStats` is used for a main account, it will `404` when looking up ironman stats. So we'd probably want to restrict it to only throwing/reporting on `HiScoresError`s. But I think it's fair to wait for a future enhancement to implement this so we don't delay this great change any longer!